docs/pages/access-controls/access-requests/oss-role-requests.mdx

```diff
@@ -0,0 +1,159 @@
+users can request access to. In this example, we will define two roles:
+
+1. `contractor`: users with this role can request elevated access to the `dba` role
+2. `dba`: this role grants access to databases
+
+There is no role for request approvers, because request approval rules can only
+be configured for Teleport Enterprise. In Open Source Teleport, approvals must
+be performed by running `tctl` on the Auth Server.
+
+**Contractor Role**
+
+Users with this role can request access to the `dba` role.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: contractor
+spec:
+  allow:
+    request:
+      roles: ['dba']
+```
+
+**DBA Role**
+
+This role grants access to databases.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: dba
+spec:
+  allow:
+    db_labels:
+      '*': '*'
+  options:
+    # Only allows the contractor to use this role for 1 hour from time of request.
+    max_session_ttl: 1h
+```
+
+## Requesting Access
+
+In OSS Teleport, requests are made from the `tsh` CLI. To create an access
+request, use the `tsh request create` command.
+
+```code
+$ tsh request create \
+    --roles=dba \
+    --reviewers=bob \
+    --reason="performing DB migration tonight"
+```
+
+By default, this command will block until the request is approved. To submit the
+request without waiting for approval, add the `--nowait` flag.
+
+Alternatively, `tsh` can automatically create an access request during the login
+process. To activate this behavior, specify the `--request-roles` flag:
+
+```code
+$ tsh login --user=alice --request-roles=dba
+# Seeking request approval... (id: bc8ca931-fec9-4b15-9a6f-20c13c5641a9)
+```
+
+This will wait for the request to be approved, and then issue credentials with
+the `dba` role automatically when the request is approved.
+
+To log in and submit the request without waiting for approval, add the
+`--request-nowait` flag. In this scenario, you will receive your regular roles
+upon login, and can elevate your access after the request is approved.
+
+```code
+# log in with an approved access request
+$ tsh login --request-id=bc8ca931-fec9-4b15-9a6f-20c13c5641a9
+```
+
+You can list requests using `tsh request ls`.
+
+```code
+$ tsh request ls
+# Token                                Requestor Metadata       Created At (UTC)    Status
+# ------------------------------------ --------- -------------- ------------------- -------
+# bc8ca931-fec9-4b15-9a6f-20c13c5641a9 alice     roles=dba      07 Nov 19 19:38 UTC PENDING
+```
+
+## Reviewing requests
+
+In OSS Teleport, Access Requests must be reviewed by a cluster administrator
+with the ability to run `tctl` on the Auth Server.
+
+Administrators can list requests with `tctl requests ls`, and view the details
+of a particular request with `tctl requests get`.
+
+To quickly approve or deny a request, use `tctl requests review`.
+
+For example, to approve a request:
+
+```code
+$ tctl requests review --approve --author=bob bc8ca931-fec9-4b15-9a6f-20c13c5641a9
+```
```

Comment:
Hey, my bad, I was misremembering the requirements for this when we chatted earlier

It is not necessary to run `tctl` on the auth server necessarily, you just need `update` permissions for `access_request` resources in one of your roles. So you could have a reviewer role like this and use tctl remotely:
```yaml
kind: role
version: v6
metadata:
  name: reviewer
spec:
  allow:
    - resources:
      - access_request
      verbs:
      - update
```

Also, I believe `tctl requests review` will not work in OSS. This is because whatever user you give for the `--author` flag, they will not be allowed to review requests, because non of their roles can have `spec.allow.review_requests` in OSS.

So, only `tctl request approve` or `tctl request deny` should work in OSS.

Reply:
I can confirm that `requests review` did not work for me, running on the auth server. But I could not add the `- resources` section to my existing reviewer role, getting `not a valid resource declaration` from the UI.

Reply:
My bad, should have actually tried that, I was missing the `rules` block in the yaml, I've updated the yaml in the original comment to a role I just tested

Reply:
Thanks for the update. I added the required lines to my `reviewer` role:

```yaml
kind: role
metadata:
  id: BLAH
  name: reviewer
spec:
  allow:
    ...
    ...
    rules:
    - resources:
      - access_request
      verbs:
      - update
    ...
    ...
```

Then I logged out and in as a user with that role:

```sh
> Profile URL:        https://myhomelabdomain
  Logged in as:       alex
  Cluster:            myhomelabdomain
  Roles:              access, auditor, dba, editor, reviewer, sp-manager
```

But I still don't see requests:

```
❯ tctl requests ls
Token Requestor Metadata Resources Created At (UTC) Request TTL Session TTL Status Request Reason Resolve Reason 
----- --------- -------- --------- ---------------- ----------- ----------- ------ -------------- -------------- 


❯
```

Reply:
right, in order to view them you will need the `read` and `list` verbs, I should have included those

Reply:
> So you could have a reviewer role like this and use tctl remotely:

I'm not sure we should even mention this. If you can't use tctl locally, then you're probably on Cloud, which means you can use tsh.

Adding these `allow` rules is a slippery slope because you can accidentally give yourself permission too many permissions and break the security of the access request system. What do you think about not mentioning it in the docs, even though it does work?

Reply:
> If you can't use tctl locally, then you're probably on Cloud

Or you're a security tech with admin rights in Teleport, but not given direct access to the host running the auth service.
Or you can access it, but it's difficult enough to access a shell in that environment. In my case, the auth service is in a container on a server, so I need to SSH then `docker exec` a shell or `tctl` command. Or what about a k8s cluster?

Just my 2¢, do as you will.

Reply:
Okay @nklaassen @alexfornuto it seems this is our unresolved item.

I am hesitant to even mention how to configure RBAC to run tctl remotely here, as these rules allow you to modify any access request, including those of your own or of others. If this is done in an enterprise environment it could result in privilege escalation attacks.

Thoughts?

Reply:
I agree that we should not recommend to include `update` rules for access requests, it gives too much power to the reviewer. Technically running `tctl` directly on Auth gives you even **more** power that could be abused, but I think it makes it clear that this should only be possible for cluster admins and avoids us publishing a dangerous config that people could copy and use without realizing it.

Reply:
Doesn't this leave the issue that `request review` doesn't work locally?